subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncherFactory.java

```diff
@@ -63,10 +64,13 @@
     private DefaultGradleLauncher rootBuild;
 
     public DefaultGradleLauncherFactory(
-        ListenerManager listenerManager, ProgressLoggerFactory progressLoggerFactory, GradleUserHomeScopeServiceRegistry userHomeDirServiceRegistry) {
+        ListenerManager listenerManager, MemoryStatusBroadcaster memoryStatusBroadcaster, ProgressLoggerFactory progressLoggerFactory, GradleUserHomeScopeServiceRegistry userHomeDirServiceRegistry) {
         this.globalListenerManager = listenerManager;
         this.userHomeDirServiceRegistry = userHomeDirServiceRegistry;
 
+        // Start broadcasting memory status updates
+        memoryStatusBroadcaster.start();
```

Comment:
Should put this somewhere more specific to memory monitoring. I'd just have `MemoryStatusBroadcaster` take care of starting itself on construction, or have `MemoryManager` start it on construction. Or just merge `MemoryStatusBroadcaster` and `MemoryManager`, I don't see any real point in them being separate.